From: Svante Signell <svante.signell@gmail.com>
Date: Tue, 16 Feb 2021 19:37:13 +0100
Subject: Don't use hrtimers on GNU Hurd

This patch solves the build error:

    PHP Warning: PHP Startup: Failed to initialize high-resolution timer in
    Unknown on line 0.
---
diff --git a/Zend/zend_hrtime.h b/Zend/zend_hrtime.h
index a67c9d210c..fcbc5e656f 100644
--- a/Zend/zend_hrtime.h
+++ b/Zend/zend_hrtime.h
@@ -37,7 +37,10 @@
 #define ZEND_HRTIME_PLATFORM_HPUX    0
 #define ZEND_HRTIME_PLATFORM_AIX     0
 
-#if defined(_POSIX_TIMERS) && ((_POSIX_TIMERS > 0) || defined(__OpenBSD__)) && defined(_POSIX_MONOTONIC_CLOCK) && defined(CLOCK_MONOTONIC)
+#if defined(_POSIX_TIMERS) && \
+    ((_POSIX_TIMERS > 0) || defined(__OpenBSD__) || defined(__GNU__)) && \
+    defined(_POSIX_MONOTONIC_CLOCK) && \
+    defined(CLOCK_MONOTONIC)
 # undef  ZEND_HRTIME_PLATFORM_POSIX
 # define ZEND_HRTIME_PLATFORM_POSIX 1
 #elif defined(_WIN32) || defined(_WIN64)
 